daml-lf/spec/daml-lf-1.rst

```diff
@@ -2979,6 +2970,16 @@ as described by the ledger model::
        ⇓ᵤ
      (Err (Fatal "Exercise on unknown contract"), ε)
 
+     'tpl' (x : T)
+         ↦ { 'choices' { …, 'choice' ChKind Ch (y : 'ContractId' Mod:T) (z : τ) : σ 'by' eₚ 'observers' eₒ ↦ eₐ, … }, … }  ∈  〚Ξ〛Mod
+     cid ∈ dom(st₀)
+     st(cid) = (Mod':T', vₜ, 'inactive')
+     Mod:T ≠ Mod':T'
+   —————————————————————————————————————————————————————————————————————— EvUpdExercWrongTemplate
+     'exercise' Mod:T.Ch cid v₁ ‖ (st; keys₀)
+       ⇓ᵤ
+     (Err (Fatal "Exercise on contract of wrong template"), ε)
+
```

Comment:
Good catch. However,  I think it should go between `EvUpdExercNestingArgErr` and `EvUpdExercBodyEvalErr`,


```suggestion
    'tpl' (x : T)
         ↦ { 'choices' { …, 'choice' ChKind Ch (y : 'ContractId' Mod:T) (z : τ) : σ 'by' eₚ 'observers' eₒ ↦ eₐ, … }, … }  ∈  〚Ξ〛Mod
     cid ∈ dom(st₀)
     st₀(cid) = (Mod':T', vₜ, 'active')
     eₚ[x ↦ vₜ, z ↦ v₁]  ⇓  Ok vₚ
     eₒ[x ↦ vₜ, z ↦ v₁]  ⇓  Ok vₒ
     |v₁| ≤ 100
     Mod:T ≠ Mod':T'
—————————————————————————————————————————————————————————————————————— EvUpdExercWrongTemplate
     'exercise' Mod:T.Ch cid v₁ ‖ (st; keys₀)
       ⇓ᵤ
     (Err (Fatal "Exercise on contract of wrong template"), ε)

```

Reply:
the line 
```
st₀(cid) = (Mod:T, vₜ, 'active')
```
should be changed by
```
st₀(cid) = (Mod':T', vₜ, 'active')
```
in the rules from `EvUpdExercInactive` to `EvUpdExercNestingArgErr` (both included).

Reply:
I'd like this rule added to `compiler/damlc/tests/daml-test-files/SemanticsEvalOrder.daml` (and a test for `EvUpdNestingArgErr` as well...). We can add these in a separate PR.

Reply:
> I'd like this rule added to compiler/damlc/tests/daml-test-files/SemanticsEvalOrder.daml (and a test for EvUpdNestingArgErr as well...). We can add these in a separate PR.

Problably not a job for @andreaslochbihler-da. Should we create an issue ?

Reply:
Good idea, here is the issue: https://github.com/digital-asset/daml/issues/11861